src/Blocks/BlockBigFlower.h

```diff
@@ -125,7 +121,7 @@ class cBlockBigFlowerHandler final :
 			auto lowerPartPos = a_BlockPos - Vector3i(0, 1, 0);
 			if (a_ChunkInterface.GetBlock(lowerPartPos) == a_OldBlockType)
 			{
-				a_ChunkInterface.DropBlockAsPickups(lowerPartPos);
```

Comment:
You could keep dropping the bottom as a pickup but that causes the creative dropping issue. So _that's_ why it happens...

I guess you'll need to check if `a_Digger->IsPlayer()` and if `static_cast<cPlayer *>(a_Digger)->IsGameModeCreative()` then `SetBlock` else `DropBlockAsPickups`.